src/osd/PrimaryLogPG.cc

```diff
@@ -10078,6 +10078,76 @@ void PrimaryLogPG::cancel_copy_ops(bool requeue, vector<ceph_tid_t> *tids)
   }
 }
 
+struct C_gather : public Context {
```

Comment:
You also need to record the current peering interval.  See C_CopyFrom::last_peering_reset.  Upon construction, use pg->get_last_peering_reset() to get the start epoch of the interval and check against it in finish.